etcdserver/server.go

```diff
@@ -632,6 +628,21 @@ func (s *EtcdServer) Lead() uint64 { return atomic.LoadUint64(&s.r.lead) }
 
 func (s *EtcdServer) Leader() types.ID { return types.ID(s.Lead()) }
 
+// RequestTimeout returns an appropriate timeout to wait for request to be committed.
+func (s *EtcdServer) RequestTimeout() time.Duration {
+	// expectedCommitDelay
+	// + 2 * heartbeat for possible message lost
+	// + 2 * election timeout for possible leader election
+	return s.expectedCommitDelay() + 2*time.Duration(s.cfg.TickMs)*time.Millisecond + 2*time.Duration(s.cfg.ElectionTicks)*time.Duration(s.cfg.TickMs)*time.Millisecond
+}
+
+func (s *EtcdServer) expectedCommitDelay() time.Duration {
```

Comment:
this is absolutely not expected commit delay. This might be the max.

The expected delay is around `100ms (disk io latency) + rtt`